javascript-carousel/main.js

```diff
@@ -0,0 +1,77 @@
+var $carousel = document.querySelector('.carousel');
+var $chevLeft = document.querySelector('.left');
+var $chevRight = document.querySelector('.right');
+var $image = document.querySelectorAll('.image');
+var $circle = document.querySelectorAll('.circle');
+
+$carousel.addEventListener('click', function () {
+  clearInterval(advanceIntervalId);
+
+  if (event.target === $chevLeft) {
+    for (var i = 0; i < $image.length; i++) {
+      if ($image[i].getAttribute('class') === 'image') {
+        $image[i].setAttribute('class', 'image hidden');
+        $circle[i].setAttribute('class', 'far fa-circle selector circle');
+        if (i === 0) {
+          $image[$image.length - 1].setAttribute('class', 'image');
+          $circle[$circle.length - 1].setAttribute('class', 'fas fa-circle selector circle');
+          break;
+        } else {
+          $image[i - 1].setAttribute('class', 'image');
+          $circle[i - 1].setAttribute('class', 'fas fa-circle selector circle');
+        }
+      }
+    }
+  } else if (event.target === $chevRight) {
+    for (var j = 0; j < $image.length; j++) {
+      if ($image[j].getAttribute('class') === 'image') {
+        $image[j].setAttribute('class', 'image hidden');
+        $circle[j].setAttribute('class', 'far fa-circle selector circle');
+        if (j === $image.length - 1) {
+          $image[0].setAttribute('class', 'image');
+          $circle[0].setAttribute('class', 'fas fa-circle selector circle');
+        } else {
+          $image[j + 1].setAttribute('class', 'image');
+          $circle[j + 1].setAttribute('class', 'fas fa-circle selector circle');
+          break;
+        }
+      }
+    }
```

Comment:
This code could be dramatically simplified if you had a data model (a variable) to store the currently active index. The logic would be something like:
1. Determine the next or previous index based on whether to increment, decrement, or "wrap" if we're already at 0 or the end.
1. loop through all images and hide all but the image at the new active index
1. loop through all dots and style all but the dot at the new active index